apps/src/lab2/levelEditors/predictSettings/EditPredictSettings.tsx

```diff
@@ -0,0 +1,119 @@
+import React, {useState} from 'react';
+import {LevelPredictSettings, PredictQuestionType} from '../types';
+import {SimpleDropdown} from '@cdo/apps/componentLibrary/dropdown';
+import Checkbox from '@cdo/apps/componentLibrary/checkbox';
+import moduleStyles from './edit-predict-settings.module.scss';
+import {BodyThreeText} from '@cdo/apps/componentLibrary/typography';
+import FreeResponseFields from './FreeResponseFields';
+import MultipleChoiceFields from './MultipleChoiceFields';
+
+interface EditPredictSettingsProps {
+  initialSettings: LevelPredictSettings;
+}
+
+const EditPredictSettings: React.FunctionComponent<
+  EditPredictSettingsProps
+> = ({initialSettings}) => {
+  const [predictSettings, setPredictSettings] = useState<LevelPredictSettings>(
+    initialSettings || {is_predict_level: false}
+  );
+
+  const handleIsPredictToggle = () => {
```

Comment:
I'm a little fuzzy on when we should use `useCallback`. The [docs](https://react.dev/reference/react/useCallback) say to use it sparingly, and this case doesn't seem to fall into the cases they talk about. Is there a reason you think it would be useful here?

Reply:
Hmm yeah I thought generally it was good practice to wrap most functions with it, but you're right that the docs say otherwise. Generally I think if the function is going to be passed as a prop to another component or used as a dependency for another hook then it's a good idea to wrap it - in this case though, the `setPredictSettings` function would be passed as a dependency if this was wrapped in a useCallback, so maybe there's some slight optimization there, but i don't think `setPredictSettings` is meant to change that often, so it's probably fine?